inst/htmlwidgets/vegawidget.js

```diff
@@ -15,47 +15,64 @@ HTMLWidgets.widget({
 
       renderValue: function(x) {
 
-        var chart_spec = x.chart_spec;
-        var embed_options = x.embed_options;
-
-        view_promise = vegaEmbed(el, chart_spec, opt = embed_options).then(function(result) {
-
-          // By removing the style (width and height) of the
-          // enclosing element, we let the "chart" decide the space it
-          // will occupy.
-          //
-          el.removeAttribute("style");
-
-          return(result.view);
-        }).catch(console.error);
-
+        // initialise promise
+        view_promise = vegaEmbed(el, x.chart_spec, opt = x.embed_options);
+
+        // fulfill promise by rendering the visualisation
+        view_promise
+          .then(function(result) {
+            // By removing the style (width and height) of the
+            // enclosing element, we let the "chart" decide the space it
+            // will occupy.
+            el.removeAttribute("style");
+          })
+          .catch(console.error);
       },
 
       resize: function(width, height) {
 
       },
 
+      // the view can just be an attribute of the HTMLWidgets object
```

Comment:
Is this comment suggesting that the getView function can probably be removed? I think that'd probably be a reasonable thing to do

Reply:
I agree - I think `getView` can be removed; I think the only usage is in `getVegaView`